src/Std/Internal/Async/Basic.lean

```diff
@@ -109,6 +329,430 @@ def getState (x : AsyncTask α) : BaseIO IO.TaskState :=
 
 end AsyncTask
 
+/--
+A `MaybeTask α` represents a computation that either:
+
+- Is immediately available as an `α` value, or
+- Is an asynchronous computation that will eventually produce an `α` value.
+-/
+inductive MaybeTask (α : Type)
+  | pure : α → MaybeTask α
+  | ofTask : Task α → MaybeTask α
+
+namespace MaybeTask
+
+/--
+Constructs an `Task` from a `MaybeTask`.
+-/
+@[inline]
+def toTask : MaybeTask α → Task α
+  | .pure a => .pure a
+  | .ofTask t => t
+
+/--
+Gets the value of the `MaybeTask` by blocking.
+-/
+@[inline]
+def wait {α : Type} (x : MaybeTask α) : α :=
```

Comment:
```suggestion
def get {α : Type} (x : MaybeTask α) : α :=
```

To avoid have two different names for what is essentially the same operation (c.f., `Task.get`).